framework/helpers/docker.py

```diff
@@ -0,0 +1,331 @@
+from collections import defaultdict
+import datetime
+import logging
+import math
+import pathlib
+import queue
+import threading
+
+from docker.client import DockerClient
+from docker.errors import NotFound
+from docker.types import ContainerConfig
```

Comment:
As discussed before, we don't import classes from files: https://google.github.io/styleguide/pyguide.html#224-decision

The only exceptions from this are:
- typing module
- collections.abc module
- typing_extensions module

Reply:
Frankly speaking, I do not quite understand all those different import types in Python. I addressed this comment as I understood it.